samples/objc/FirebaseUI-demo-objc/Samples/Auth/FIRAuthViewController.m

```diff
@@ -81,11 +81,13 @@ - (void)viewWillAppear:(BOOL)animated {
     [weakSelf updateUI:auth withUser:user];
   }];
 
+  [self.navigationController setToolbarHidden:NO animated:YES];
 }
 
--(void)viewDidDisappear:(BOOL)animated {
-  [super viewDidDisappear:animated];
+-(void)viewWillDisappear:(BOOL)animated {
+  [super viewWillDisappear:animated];
   [self.auth removeAuthStateDidChangeListener:self.authStateDidChangeHandle];
+  [self.navigationController setToolbarHidden:YES animated:YES];
```

Comment:
You should probably pass the `animated` flag through to `setToolbarHidden:animated:`.

```objc
[self.navigationController setToolbarHidden:YES animated:animated];
```

That way your code doesn't create animations if they're not going to be visible anyway.